systems/plants/RigidBodyManipulator.h

```diff
@@ -212,6 +212,8 @@ class DLLEXPORT_RBM RigidBodyManipulator
   DrakeCollision::ElementId addCollisionElement(const RigidBody::CollisionElement& element, const std::shared_ptr<RigidBody>& body, std::string group_name);
 
   void updateCollisionElements(const std::shared_ptr<RigidBody>& body);
+  void getTerrainPoints(int body_id, Eigen::MatrixXd &terrain_points);
```

Comment:
I agree with your argument about `Eigen::Matrix3Xd` and change the code accordingly.